pilot/pkg/model/monitoring.go

```diff
@@ -0,0 +1,33 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import "istio.io/pkg/monitoring"
+
+func init() {
+	monitoring.MustRegister(
+		providerLookupClusterFailures,
+	)
+}
+
+var providerLookupClusterFailures = monitoring.NewSum(
```

Comment:
I think Gauge is better, we can record which cluster look up failed rather than always increasing the num

Reply:
when should pilot decrease the count?

Reply:
Please refer to envoyfilter patching metric, the sum number here is meaningless, it oculd be unlimited big

Reply:
For a counter, user should care about `rate` or `increase`.
if you are talking about `pilot_envoy_filter_status`, it's disabled by default.

Reply:
cc @ramaraochavali , do you know why it's disabled?

Reply:
It is disabled because of possible performance impact. People who want will only enable it in dev envs

Reply:
After take a look at `istio.io/pkg/monitoring`, it's impossible to reset all metrics to `0` when services or telemetries chaged.
cc @hzxuzhonghu